### PR TITLE
CamelCase Completion For Indexed Resources and PC 

### LINKED ISSUE #1003: FLX and CamelCase Completion
Hello. Ensime is looking pretty good. Thanks! Editing Java in Eclipse, I'm used to having camel case completion. I'm sure you know what I mean.... Is there a way to set that for use with Ensime?

### PR DESCRIPTION
**Pull request by hzenginx**
close #1003

Introduce a wildcard query to support camel case completion for indexed resources.
Tweak PC prefix filter to support camel case completion.

### CODE REVIEW

**Comment by fommil** [comment]
looks like you used a different email address for this commits so github can't identify them to your account.

**Comment by hzenginx** [comment]
@fommil looks like my git config was using my company email, fixed and pushed 👍

File: core/src/main/scala/org/ensime/core/Completion.scala [hunk 1]
```diff
@@ -376,9 +376,12 @@ object CompletionUtil {
     matchEntire: Boolean, caseSens: Boolean): Boolean = {
     val prefixUpper = prefix.toUpperCase
 
+    val prefixRegexp = "[A-Za-z0-9]*" + prefix.replaceAll("(?<!^)([A-Z])", "[A-Za-z0-9]*$1") + "[A-Za-z0-9]*"
+
     (matchEntire && m == prefix) ||
```
Comment (fommil) [inline core/src/main/scala/org/ensime/core/Completion.scala]: nice. I keep meaning to allow the exclusion of a bunch of noise in here. using a custom `sysdef` (typelevel scala) helps a **lot**.

File: core/src/main/scala/org/ensime/core/Completion.scala [hunk 2]
```diff
@@ -376,9 +376,12 @@ object CompletionUtil {
...
       (!matchEntire && caseSens && m.startsWith(prefix)) ||
-      (!matchEntire && !caseSens && m.toUpperCase.startsWith(prefixUpper))
+      (!matchEntire && !caseSens && m.toUpperCase.startsWith(prefixUpper)) ||
+      (!matchEntire && prefix.filter(_.isUpper).nonEmpty && m.matches(prefixRegexp))
```
Comment (fommil) [inline core/src/main/scala/org/ensime/core/Completion.scala]: `.exists` may be marginally more efficient than `filter` followed by `nonEmpty`

**Comment by fommil** [comment]
looks like you're getting different hits for search depending on the OS. The test may be overspecifying the requirement.

**Comment by hzenginx** [comment]
I am trying to reproduce the problem, may be scalatest causing this IDK yet

**Comment by fommil** [comment]
the ordering of search results is subjective to IO orderings (as well as classpath orders across scala / java versions) we've seen similar things in the past.

**Comment by hzenginx** [comment]
I replaced the filter+nonEmpty with exists.
Test was too general; I tweaked the prefix with a more specific one.

**Comment by fommil** [comment]
thanks! This is the deploy job https://ci.fommil.com/ensime/ensime-server/267

**Pull request merged**